### ISSUE TITLE
BOM (Byte-order-mark) crashes parser

### ISSUE THREAD

**Issue by siedentop** [cheshirekow/cmake_format#79]
Thank you for this amazing tool! 

I tried this on a bigger code-base, and discovered that the parser fails in line 580 (`parser.py`), if there is a Byte-Order-Mark bytes at the beginning of the file.

The function which asserts is `consume_body` and it fails because the "switch" statement does not know have a matching TokenType

I am aware of your statement in #57 that you do not want to succeed on invalid CMake files. However, according to Wikipedia, UTF-8 files may use BOM [1].

## Details
`Token(type=UNQUOTED_LITERAL, content=u'\ufeffcmake_minimum_required', line=1, col=0)`

```
Traceback (most recent call last):
  File "/path/to/venv/bin/cmake-format", line 9, in <module>
    load_entry_point('cmake-format==0.4.3', 'console_scripts', 'cmake-format')()
  File ".../cmake_format/cmake_format/__main__.py", line 336, in main
    process_file(cfg, infile, outfile, args.dump)
  File ".../cmake_format/cmake_format/__main__.py", line 59, in process_file
    parse_tree = parser.parse(tokens, config.fn_spec)
  File ".../cmake_format/cmake_format/parser.py", line 600, in parse
    return consume_body(tokens, cmdspec)
  File ".../cmake_format/cmake_format/parser.py", line 584, in consume_body
    tokens[0].begin.line, tokens[0].begin.col))
AssertionError: Unexpected UNQUOTED_LITERAL token at 1:0
```

## Possible solutions
* Would it be acceptable to treat `\ufeff` as a whitespace token? The downside is that it probably would get deleted.
* Don't fix. (I have now simply removed the BOM from the offending files.)
* Something else?

[1]: https://en.wikipedia.org/wiki/Byte_order_mark

---
* I used newest master commit.  (161a19b9b)
* Ubuntu 14.04
* Python 2.7

**Comment by cheshirekow** [comment]
If it's valid utf-8 cmake-for at should probably not crash out. Since the goal of the tool is to get consistent formatting, I think I should parse the token, ignore it on input, and have an option of whether or not to include it on output.

I'm not familiar with a byte-order mark though and I'm under the impression that utf-8 doesn't have a byte-order. Is this a utf-16 thing? I will read about it and get back.

**Comment by unapiedra** [comment]
I probably are ahead of you in knowledge about byte-order-mark by about 30 minutes. So, no help about this, I am sorry.

**Comment by cheshirekow** [comment]
This looks like a pretty good answer: https://stackoverflow.com/a/2223926

Sounds like it's valid utf-8 but not recommended to be used. I think what I'll do is: lex the token (don't crash on input), drop it for output by default, add an option to emit BOM for people who want it on output.

**Comment by cheshirekow** [comment]
Ok, should be fixed in `0.4.4`. I added a test. hopefully it is representative.
